src/solver/conv_direct_naive_conv.cpp

```diff
@@ -108,7 +110,15 @@ bool IsOutputInt32(const ProblemDescription& problem)
 std::string ConvDirectNaiveConvKernelName(const ProblemDescription& problem)
 {
     std::ostringstream kernel_name;
-    kernel_name << "naive_conv_";
+    if(miopen::IsEnvvarValueEnabled("MIOPEN_USE_PACKED_CONV_REF_KERNEL"))
```

Comment:
```suggestion
    if(miopen::IsEnabled("MIOPEN_DEBUG_CONV_DIRECT_NAIVE_USE_PACKED_KERNELS"))
```
Do not use `IsEnvvarValueEnabled()`.

Declare env var 
```C++
MIOPEN_DECLARE_ENV_VAR(MIOPEN_DEBUG_CONV_DIRECT_NAIVE_USE_PACKED_KERNELS)
```
somewhere above.

Reply:
I implemented your suggestion but I don't see the benefit of this approach as  `IsEnabled` does not seem to add any benefit  over `IsEnvvarValueEnabled`.

Reply:
```C++
template <class T>
inline bool IsEnabled(T)
{
    static const bool result = miopen::IsEnvvarValueEnabled(T::value());
    return result;
}
```
Please consider the effect of `static const` and you'll see the benefit.
[resolved]

Reply:
Also it's better to define and use env variable in a common way across the project. For example, you can always get all the possible variables by `git grep MIOPEN_DECLARE_ENV_VAR`.

Reply:
OK, agree with both of you. I would have cached the `bool result` inside the environment variable's class `T`, but this is equally good.